libs/ui-toolkit/src/components/callout/callout.tsx

```diff
@@ -1,23 +1,62 @@
+import type { ReactNode } from 'react';
 import classNames from 'classnames';
 import { getIntentShadow, Intent } from '../../utils/intent';
 import type { IconName } from '../icon';
 import { Icon } from '../icon';
 
-export interface CalloutProps {
+interface CalloutRootProps {
   children?: React.ReactNode;
   title?: React.ReactElement | string;
   intent?: Intent;
-  iconName?: IconName;
   headingLevel?: 1 | 2 | 3 | 4 | 5 | 6;
 }
 
-export function Callout({
-  children,
-  title,
-  intent = Intent.Help,
-  iconName,
-  headingLevel,
-}: CalloutProps) {
+interface CalloutWithoutIcon extends CalloutRootProps {
+  iconName?: never;
+  icon?: never;
+}
+
+interface CalloutPropsWithIconName extends CalloutRootProps {
```

Comment:
Note (non-blocking, see below for further (blocking) refactor comments): You're extending `CalloutRootProps`, so the only props you need in here are `iconName` and `icon`. Same for `CallourPropsWithIcon`.

Reply:
i've replied to this below, as it's about the exact same topic as your other comment i believe.

Reply:
No, this is an implementation detail - you're extending `CalloutRootProps`, but redefining the same props here (`headingLevel`, `title` etc). You do the same with `CalloutPropsWithIcon`. You've done it properly on  `CalloutWithoutIcon` however.

Reply:
i see, sorry, missed that part. totally right, removed the redundancies.